tests/docker/Dockerfile

```diff
@@ -242,7 +242,7 @@ RUN /polaris && rm /polaris
 #################################
 
 FROM base as iceberg-rest
-COPY --chown=0:0 tests/java/iceberg-rest /opt/iceberg-rest-catalog
+COPY --chown=0:0 tests/java/iceberg-rest /opt/redpanda-tests/java/iceberg-rest-catalog
```

Comment:
>a tiny fixup to make it easy to integrate into vtools, no functional changes.

I'm curious what this means though -- does the vtools task not work without this change? Or is it mostly cosmetic?

Reply:
cdt would fail (they will look for spark/trino which are new and not hooked up yet but also would fail at iceberg-rest-catalog due to this incompat change I made.)